stonesoup/updater/pointprocess.py

```diff
@@ -0,0 +1,215 @@
+# -*- coding: utf-8 -*-
+from abc import abstractmethod
+
+from scipy.stats import multivariate_normal
+
+from ..base import Base, Property
+from .kalman import KalmanUpdater
+from ..types.update import GaussianMixtureUpdate
+from ..types.state import TaggedWeightedGaussianState
+from ..types.numeric import Probability
+
+
+class PointProcessUpdater(Base):
+    r"""
+    Base updater class for the implementation of any Gaussian Mixture (GM)
+    point process derived multi target filters such as the
+    Probability Hypothesis Density (PHD),
+    Cardinalised Probability Hypothesis Density (CPHD) or
+    Linear Complexity with Cumulants (LCC) filters
+    """
+    updater = Property(
+        KalmanUpdater,
+        doc="Underlying updater used to perform the \
+             single target Kalman Update.")
+
+    clutter_spatial_density = Property(
+        float,
+        default=1e-26,
+        doc="Spatial density of the clutter process uniformly\
+             distributed across the state space.")
+
+    normalisation = Property(
+        bool,
+        default=True,
+        doc="Flag for normalisation")
+
+    prob_detection = Property(
+        Probability,
+        default=1,
+        doc="Probability of a target being detected at the current timestep")
+
+    prob_survival = Property(
+        Probability,
+        default=1,
+        doc="Probability of a target surviving until the next timestep")
+
+    def update(self, hypotheses):
+        """
+        Updates the current components in a
+        :state:`GaussianMixture` by applying the underlying \
+        :class:`KalmanUpdater` updater to each component \
+        with the supplied measurements.
+
+        Parameters
+        ==========
+        hypotheses : list of :class:`MultipleHypothesis`
+            Measurements obtained at time :math:`k+1`
+
+        Returns
+        =======
+        updated_components : :state:`GaussianMixtureUpdate`
+            GaussianMixtureMultiTargetTracker with updated \
+            components at time :math:`k+1`
+        """
+        updated_components = list()
+        weight_sum_list = list()
+        # Loop over all measurements
+        for multi_hypothesis in hypotheses[:-1]:
+            updated_measurement_components = list()
+            # Initialise weight sum for measurement to clutter intensity
+            weight_sum = 0
+            # For every valid single hypothesis, update that component with
+            # measurements and calculate new weight
+            for hypothesis in multi_hypothesis:
+                measurement_prediction = \
+                    self.updater.predict_measurement(
+                            hypothesis.prediction)
```

Comment:
`predict_measurement` also needs `hypothesis.measurement.measurement_model` passing to it.